### DISCUSSION TITLE
`Delete` and `Replicate` doesn't function as expected

### DISCUSSION THREAD

**Discussion by bilogic** [filamentphp/filament#8742] [Help]
https://gist.github.com/bilogic/3c8d9bb37e3a81e319fb2e5b8b8d364a

1. I added a form to a page
2. `Save` works as expected, triggering a series of methods leading to `handleRecordUpdate`
3. However, `Delete` doesn't run its `->action(...)`
4. `Replicate` also does not redirect to `/admin`

Any idea what I might be missing?

**Comment by VincvanAs** [comment]
For delete you can use:

DeleteAction::make()
    ->before(function () {
        // ...
    })
    ->after(function () {
        // ...
    })

Or just simply ->successNotificationTitle('Row deleted')

For the replica action, " i think " you need to define a route e.g. getUrl()

ReplicateAction::make()
    ->successRedirectUrl(route('posts.list')

More information on Actions:
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create

**Reply by bilogic** [reply]
I tried `>before(...)`, `>after(...)` and `>successNotificationTitle(...)`, the spinner appears in front of the `Delete` button, then disappears and nothing else happens. 

Do note, I'm not trying to delete anything, just trying to understand how these buttons work and how I can reuse or make similar buttons do other things.

**Reply by VincvanAs** [reply]
Delete and Replicate are predefined actions.

I think i see the "issue" now.

In order to call a predefined action such as Delete or Replicate you need to define the ->record($this->record).

**Reply by bilogic** [reply]
ok, what if I just want a simple action? which namespace should I use?

This did not work too:
```PHP
use Filament\Pages\Actions\Action;

        $actions[] = Action::make($form.'DeleteAction')
            ->hidden(false)
            ->successNotificationTitle('Row deleted')
            ->action(function () {
                return Notification::make()
                    ->title('asdfasdfas')
                    ->body('asdfasdf')
                    ->color('success')
                    ->send();
            });
```

**Reply by VincvanAs** [reply]
I recommend you to read the documentation about actions https://filamentphp.com/docs/3.x/actions/overview

**Reply by bilogic** [reply]
I did and it got me nowhere even after trial and error, that's why I created the gist and this discussion.
